spring-boot/configuration/src/main/java/io/reflectoring/validation/ReportEmailAddressValidator.java

```diff
@@ -0,0 +1,27 @@
+package io.reflectoring.validation;
+
+import org.springframework.validation.Errors;
+import org.springframework.validation.ValidationUtils;
+import org.springframework.validation.Validator;
+
+class ReportEmailAddressValidator implements Validator {
```

Comment:
Is there a test that shows how this validator works when an email is invalid?

Reply:
There is a test method in [AppPropertiesInvalidInputTest](https://github.com/thombergs/code-examples/pull/26/commits/735d80d72fa1cd1944d768bfa0654131d203a073#diff-3f59dfb62edad46b481cfa1bcd6bed9aR92) if you mean checking the email not to contain the specified domain. But this validator doesn't check the format of the email, only checks if it is not empty and contains the given domain. 

The format of the email field actually checked by _@Email_ annotation before this validator. So if we pass bean validation's _@Email_ check somehow, then the input for example "@analysisapp.com" would pass this validator. Was that what you meant actually?